handlers/login.go

```diff
@@ -226,7 +234,7 @@ func getValidRequestedURL(r *http.Request) (string, error) {
 	return u.String(), nil
 }
 
-func oauthLoginURL(r *http.Request, state string) string {
+func oauthLoginURL(session sessions.Session, r *http.Request, state string) string {
```

Comment:
- `session` should be after `r` in the signature
- drop `state` and adjust the function to pull `state` from `session`